src/GridFS/CollectionWrapper.php

```diff
@@ -80,6 +80,30 @@ public function deleteChunksByFilesId($id): void
         $this->chunksCollection->deleteMany(['files_id' => $id]);
     }
 
+    /**
+     * Delete all GridFS files and chunks for a given filename.
+     */
+    public function deleteFileAndChunksByFilename(string $filename): int
+    {
+        /** @var iterable<array{_id: mixed}> $files */
+        $files = $this->findFiles(['filename' => $filename], [
+            'codec' => null,
```

Comment:
This shouldn't be necessary.

Operations such as Find already raise exceptions if both a `codec` _and_ `typeMap` are specified, and `Collection::find()` relies on `inheritCodecOrTypeMap()`, which avoids inheriting `codec` if an explicit `typeMap` option is specified (and vice versa).